### PR TITLE
Image Segmenter node

### PR DESCRIPTION
**Pull request by kastentx**
These changes add the image segmenter node, which supports both methods offered by the MAX model.

Default values, icons and colors are all consistent with existing nodes.

Also updated the package README and info sidebar with relevant documentation.

### CODE REVIEW

File: image-segmenter/test/node_spec.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        });
+    });
+
+    it('should handle get_metadata()', function (done) {
+        var flow = [
+            { id: 'n1', type: 'image-segmenter', name: 'image-segmenter',
+                method: 'get_metadata',
+                wires: [['n3']],
+                service: 'n2' },
+            { id: 'n2', type: 'image-segmenter-service', host: 'http://<host name>' }, // (4) define host name
```
Comment (ptitzler) [inline image-segmenter/test/node_spec.js]: Replace `<host name>` with the URL for the long running instance. This causes the test case to fail.

File: image-segmenter/test/node_spec.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        });
+    });
+    it('should handle predict()', function (done) {
+        var flow = [
+            { id: 'n1', type: 'image-segmenter', name: 'image-segmenter',
+                method: 'predict',
+                predict_image: '<node property>', // (1) define node properties
+                wires: [['n3']],
+                service: 'n2' },
+            { id: 'n2', type: 'image-segmenter-service', host: 'http://<host name>' }, // (4) define host name
```
Comment (ptitzler) [inline image-segmenter/test/node_spec.js]: Replace `<host name>` with the URL for the long running instance. This causes the test case to fail.

File: image-segmenter/test/node_spec.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+            var n1 = helper.getNode('n1');
+            n3.on('input', function (msg) {
+                try {
+                    msg.should.have.property('payload', '<output message>'); // (3) define output message
+                    done();
+                } catch (e) {
+                    done(e);
+                }
+            });
+            n1.receive({ payload: '<input message>' }); // (2) define input message
```
Comment (ptitzler) [inline image-segmenter/test/node_spec.js]: Replace placeholder

File: image-segmenter/test/node_spec.js [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+                service: 'n2' },
+            { id: 'n2', type: 'image-segmenter-service', host: 'http://<host name>' }, // (4) define host name
+            { id: 'n3', type: 'helper' }
+        ];
+        helper.load(node, flow, function () {
+            var n3 = helper.getNode('n3');
+            var n1 = helper.getNode('n1');
+            n3.on('input', function (msg) {
+                try {
+                    msg.should.have.property('payload', '<output message>'); // (3) define output message
```
Comment (ptitzler) [inline image-segmenter/test/node_spec.js]: Replace placeholder

File: image-segmenter/test/node_spec.js [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
```
Comment (ptitzler) [inline image-segmenter/test/node_spec.js]: Replace placeholder

Comment (kastentx) [reply image-segmenter/test/node_spec.js]: Tests! I will add those in and push again with that completed.

**Comment by ptitzler** [comment]
Closes #30

**Comment by kastentx** [comment]
@ptitzler I've noticed that `get_metadata` is now failing for `human-pose-estimator` and `inception_resnet_v2` because I think the response has been updated since the tests were created. The tests for `image-segmenter` are passing.

**Comment by kastentx** [comment]
I can create a new PR and/or issue for the failing `get_metadata` tests, assuming they should be passing with what comes back now.

**Comment by ptitzler** [comment]
Yes, please. We should probably refactor all metadata tests to only test for the existence of certain properties and the `id` value to minimize the impact.

**Pull request merged**